ddl/index.go

```diff
@@ -704,25 +739,67 @@ func doReorgWorkForCreateIndexMultiSchema(w *worker, d *ddlCtx, t *meta.Meta, jo
 
 func doReorgWorkForCreateIndex(w *worker, d *ddlCtx, t *meta.Meta, job *model.Job,
 	tbl table.Table, indexInfo *model.IndexInfo) (done bool, ver int64, err error) {
-	bfProcess := pickBackfillType(job)
+	bfProcess := pickBackfillType(w, job)
 	if !bfProcess.NeedMergeProcess() {
 		return runReorgJobAndHandleErr(w, d, t, job, tbl, indexInfo, false)
 	}
 	switch indexInfo.BackfillState {
 	case model.BackfillStateRunning:
-		done, ver, err = runReorgJobAndHandleErr(w, d, t, job, tbl, indexInfo, false)
-		if err != nil {
-			return false, ver, errors.Trace(err)
-		}
-		if done {
-			indexInfo.BackfillState = model.BackfillStateReadyToMerge
+		switch bfProcess {
+		case model.ReorgTypeLitMerge:
+			logutil.BgLogger().Info("[ddl] index backfill state running")
+			bc, ok := ingest.LitBackCtxMgr.Load(job.ID)
+			if ok && bc.Done() {
+				break
+			}
+			if !ok && job.SnapshotVer != 0 {
+				// The owner is crashed or changed, we need to restart the backfill.
+				job.SnapshotVer = 0
+				return false, ver, nil
+			}
+			bc, err = ingest.LitBackCtxMgr.Register(w.ctx, indexInfo.Unique, job.ID, job.ReorgMeta.SQLMode)
+			if err != nil {
+				tryFallbackToTxnMerge(job, err)
```

Comment:
This function log "txn-merge", but this function is "lit-merge". Do we need to update this log?

Reply:
If there is an error for 'lit-merge', we will fallback to 'txn-merge'.

Reply:
Got it.